geojson_pydantic/features.py

```diff
@@ -1,25 +1,26 @@
 """pydantic models for GeoJSON Feature objects."""
 
 import json
-from typing import Dict, Generic, List, Optional, TypeVar, Union
+from typing import Any, Dict, Generic, Iterator, List, Optional, Type, TypeVar, Union
 
-from pydantic import Field, ValidationError, validator
+from pydantic import BaseModel, Field, ValidationError, validator
 from pydantic.generics import GenericModel
 
 from geojson_pydantic.geometries import GeoInterfaceMixin, Geometry, GeometryCollection
 from geojson_pydantic.types import BBox
 
-Props = TypeVar("Props", bound=Dict)
-Geom = TypeVar("Geom", bound=Optional[Union[Geometry, GeometryCollection]])
+Props = TypeVar("Props", bound=Union[Dict[str, Any], BaseModel])
+Geom = TypeVar("Geom", bound=Union[Geometry, GeometryCollection])
+F = TypeVar("F", bound="Feature")
 
 
 class Feature(GenericModel, Generic[Geom, Props], GeoInterfaceMixin):
     """Feature Model"""
 
-    type: str = Field("Feature", const=True)
-    geometry: Geom = None
-    properties: Optional[Props]
-    id: Optional[str]
+    type: str = Field(default="Feature", const=True)
+    geometry: Optional[Geom] = None
```

Comment:
this has been discussed already and `geometry` is not really Optional but should at minimum set to `None` 🤷

Reply:
I think they have made things clearer in more recent typing and could look into a better way to do it with `typing_extensions` but `Optional[Geom] == Union[Geom, None]` and doesn't say the argument itself is optional. If the argument should be required then would need to remove the `= None` part.

Reply:
Looks like there are a few options to allow `None` but require the field in inputs:
```python
geometry: Union[Geom, None]
geometry: Optional[Geom] = ...
geometry: Optionsl[Geom] = Field(...)
```
I think the first is probably the most clear but happy to do whatever. 

I tried to do `geometry: Geom | None` first but cannot use it supporting older versions of python as Pydantic does not support `|` unless you are on 3.10 https://github.com/samuelcolvin/pydantic/pull/2609

Reply:
Played with it more and only way to keep all the type checking happy is with `= Field(...)` if needs to be treated as a required field. 

>    A Feature object has a member with the name "geometry".  The value
      of the geometry member SHALL be either a Geometry object as
      defined above or, in the case that the Feature is unlocated, a
      JSON null value.

The way I read that I would call it optional in the context of this module. The field exists and a default `None` seems fine. The way the code worked before it would already have defaulted to `None` without an input. So switching to a required field would be a breaking change and I don't think it would be any more correct.